packages/unpublished-adapter-encapsulation-test-app/tests/integration/queries-test.js

```diff
@@ -0,0 +1,222 @@
+        assert.equal(snapshot.length, 0, 'snapshot is passed to findAll represnts empty array');
+
+        return resolve(expectedResult);
+      }
+    }
+
+    owner.register('adapter:application', TestFindAllAdapter);
+
+    let manyArray = await store.findAll('person');
+
+    // This code is a workaround for issue https://github.com/emberjs/data/issues/6758
+    // expectedResult is unexpectedly mutated during store.findAll
+    // if IDENTIFIERS is turned on
+    expectedResult.data.forEach(result => {
+      delete result.lid;
+    });
+
+    let result = manyArray.toArray().map(person => person.serialize());
+    expectedResult = expectedResult.data.map(person => ({ data: person }));
+
+    assert.equal(findAllCalled, 1, 'findAll is called once');
+    assert.deepEqual(result, expectedResult, 'findAll returns expected result');
+  });
+
+  test('store.queryRecord calls adapter.queryRecord w/correct args', async function(assert) {
+    let queryRecordCalled = 0;
+    let expectedResult = {
+      data: {
+        id: '12',
+        type: 'person',
+        attributes: {
+          firstName: 'Gaurav',
+          lastName: 'Munjal',
+        },
+      },
+    };
+    let { owner } = this;
+    let store = owner.lookup('service:store');
+
+    class TestQueryRecordAdapter extends EmberObject {
+      queryRecord(passedStore, type, query) {
+        queryRecordCalled++;
+
+        assert.equal(passedStore, store, 'instance of store is passed to queryRecord');
+        assert.equal(type, Person, 'model is passed to queryRecord');
+        assert.deepEqual(query, { firstName: 'Gaurav' }, 'query is passed to queryRecord');
+
+        return resolve(expectedResult);
+      }
+    }
+
+    owner.register('adapter:application', TestQueryRecordAdapter);
+
+    let record = await store.queryRecord('person', { firstName: 'Gaurav' });
+
+    assert.equal(queryRecordCalled, 1, 'queryRecord is called once');
+    assert.deepEqual(record.serialize(), expectedResult, 'queryRecord returns expected result');
+  });
+
+  test('store.query calls adapter.query w/correct args', async function(assert) {
+    let queryCalled = 0;
+    let expectedResult = {
+      data: [
+        {
+          id: '14',
+          type: 'person',
+          attributes: {
+            firstName: 'Chris',
+            lastName: 'Tse',
+          },
+        },
+        {
+          id: '19',
+          type: 'person',
+          attributes: {
+            firstName: 'Chris',
+            lastName: 'Thoburn',
+          },
+        },
+      ],
+    };
+    let { owner } = this;
+    let store = owner.lookup('service:store');
+
+    class TestQueryAdapter extends EmberObject {
+      query(passedStore, type, query) {
+        queryCalled++;
+
+        assert.equal(passedStore, store, 'instance of store is passed to query');
+        assert.equal(type, Person, 'model is passed to query');
+        assert.deepEqual(query, { firstName: 'Chris' }, 'query is passed to query');
```

Comment:
While not implemented by the default adapters, the full signature of query is 5 args

```ts
query(Store: Store, schema: ModelClass, query, recordArray, options): Promise<unknown>;
```